### PR TITLE
Charlesmchen/corrupt thread view

### PR DESCRIPTION
**Pull request by charlesmchen-signal**
PTAL @michaelkirk

### CODE REVIEW

File: Signal/src/ViewControllers/HomeView/HomeViewController.m [hunk 1]
```diff
@@ -880,6 +880,11 @@ - (TSThread *)threadForIndexPath:(NSIndexPath *)indexPath
                                                                                  withMappings:self.threadMappings];
     }];
 
+    if (![thread isKindOfClass:[TSThread class]]) {
```
Comment (charlesmchen-signal) [inline Signal/src/ViewControllers/HomeView/HomeViewController.m]: First fix: Repair corrupt thread view.

File: SignalServiceKit/src/Storage/OWSStorage.m [hunk 2]
```diff
@@ -515,6 +515,20 @@ + (void)incrementVersionOfDatabaseExtension:(NSString *)extensionName
 {
     OWSLogError(@"%@", extensionName);
 
+    // Don't increment version of a given extension more than once
+    // per launch.
```
Comment (charlesmchen-signal) [inline SignalServiceKit/src/Storage/OWSStorage.m]: DRY up the debounce of db extension version increment.

File: Signal/src/ViewControllers/ConversationView/ConversationViewController.m [hunk 3]
```diff
@@ -4905,11 +4905,9 @@ - (BOOL)reloadViewItems
     }];
 
     // Flag to ensure that we only increment once per launch.
-    static BOOL hasIncrementedDatabaseView = NO;
-    if (hasError && !hasIncrementedDatabaseView) {
+    if (hasError) {
         OWSLogWarn(@"incrementing version of: %@", TSMessageDatabaseViewExtensionName);
         [OWSPrimaryStorage incrementVersionOfDatabaseExtension:TSMessageDatabaseViewExtensionName];
-        hasIncrementedDatabaseView = YES;
```
Comment (charlesmchen-signal) [inline Signal/src/ViewControllers/ConversationView/ConversationViewController.m]: No longer necessary due to second change.

File: SignalServiceKit/src/Storage/OWSStorage.m [hunk 4]
```diff
@@ -515,6 +515,20 @@ + (void)incrementVersionOfDatabaseExtension:(NSString *)extensionName
...
+    static NSMutableSet<NSString *> *incrementedViewSet = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        incrementedViewSet = [NSMutableSet new];
+    });
+    @synchronized(incrementedViewSet) {
+        if ([incrementedViewSet containsObject:extensionName]) {
+            OWSLogInfo(@"Ignoring redundant increment: %@", extensionName);
```
Comment (michaelkirk-signal) [inline SignalServiceKit/src/Storage/OWSStorage.m]: missing early return here?

Comment (charlesmchen-signal) [reply SignalServiceKit/src/Storage/OWSStorage.m]: Done, thanks.

**Comment by michaelkirk** [comment]
LGTM

**Pull request merged**